text/RFC-0007-System Collator Selection.md

```diff
@@ -0,0 +1,162 @@
+
+### Requirements
+
+- System MUST have at least one valid collator for every chain.
+- System MUST allow anyone to become a collator, provided they `reserve`/`hold` enough DOT.
+- System SHOULD select a set of collators with reasonable expectation that the set will not collude
+  to censor any subset of transactions.
+- Collators selected by governance SHOULD have a reasonable expectation that the Treasury will
+  reimburse their operating costs.
+
+## Stakeholders
+
+- Infrastructure providers (people who run validator/collator nodes)
+- Polkadot Treasury
+
+## Explanation
+
+This protocol builds on the existing
+[Collator Selection pallet](https://github.com/paritytech/cumulus/tree/b15da70/pallets/collator-selection)
+and its notion of Invulnerables. Invulnerables are collators (identified by their `AccountId`s) who
+will be selected as part of the collator set every session. Operations relating to the management
+of the Invulnerables are done through privileged, governance origins. The implementation should
+maintain an API for adding and removing Invulnerable collators.
+
+In addition to Invulnerables, there are also open slots for "Candidates". Anyone can register as a
+Candidate by placing a fixed bond. However, with a fixed bond and fixed number of slots, there is
+an obvious selection problem: The slots fill up without any logic to replace their occupants.
+
+This RFC proposes that the collator selection protocol allow Candidates to increase (and decrease)
+their individual bonds, sort the Candidates according to bond, and select the top `N` Candidates.
+The selection and changeover should be coordinated by the session manager.
+
+A FRAME pallet already exists for sorting ("bagging") "top N" groups, the
+[Bags List pallet](https://github.com/paritytech/substrate/blob/5032b8d/frame/bags-list/src/lib.rs).
+This pallet's `SortedListProvider` should be integrated into the session manager of the Collator
+Selection pallet.
+
+Despite the lack of apparent economic incentives (i.e., inflation), several reasons exist why one
+may want to bond funds to participate in the Candidates election, for example:
+
+- They want to build credibility to be selected as Invulnerable;
+- They want to ensure availability of an application, e.g. a stablecoin issuer might run a collator
+  on Asset Hub to ensure transactions in its asset are included in blocks;
+- They fear censorship themselves, e.g. a voter might think their votes are being censored from
+  governance, so they run a collator on the governance chain to include their votes.
+
+Unlike the fixed-bond mechanism that fills up its Candidates, the election mechanism ensures that
+anyone can join the collator set by placing the `Nth` highest bond.
+
+### Set Size
+
+In order to achieve the requirements listed under _Motivation_, it is reasonable to have
+approximately:
+
+- 20 collators per system chain,
+- of which 15 are Invulnerable, and
```

Comment:
any explanations of those numbers?
I would expect low number of invulnerable as it needs to be managed by governance and therefore high bar of entry.

Reply:
The numbers were recommendations from SR Labs, however I find them quite reasonable. Of course it will take time to get this many collators, and there may be situations where it makes sense to have more collators on a particular chain. But I think these are good general purpose targets.

With respect to the high bar of entry, that's the point. Due to the high number (supply) of system parachains, and low profitability (demand) for being a collator, the open election would likely be a very low bar of entry, making it vulnerable to malicious takeover. 

As the Motivation section says,

> "Another problem with economic scalability relates to the increasing number of system chains, and corresponding increase in need for collators (i.e., increase in collator slots). "Good" (highly available, non-censoring) collators will not want to compete in elections on many chains when they could use their resources to compete in the more profitable validator election. Such dilution decreases the required bond on each chain, leaving them vulnerable to takeover by hostile collator groups."

Reply:
ok I see the reasons now. However this to me is more of a short term workaround that there is not enough incentive to be system chain collector. Any plan to fix that? Relying on governance to gate keep things should be a last resort.

Reply:
I don't see governance as a _last_ resort, just as one of many tools. Our initial discussion was around creating incentives (see [forum](https://forum.polkadot.network/t/economic-model-for-system-para-collators/1010)), but it's really not scalable. We expect many more collators than validators in the system, and as you know validators are expensive (e.g. 50 system chains x 20 collators = 1000 collators). Likewise, collation is not a secure activity and doesn't warrant large incentives.

The incentives for governance-selected collators is often around building reputation as, for example, a validator.

Also from the Motivation:

> "While the network as a whole uses staking (and inflationary rewards) to attract validators, collators face different challenges in scale and have lower security assumptions than validators. Regarding scale, there exist many system chains, and it is economically expensive to pay collators a premium. Likewise, any staked DOT for collation is _not_ staked for validation. Since collator sets do not need to meet Byzantine Fault Tolerance criteria, staking as the primary mechanism for collator selection would remove stake that is securing BFT assumptions, making the network less secure."